python/ray/serve/_private/deployment_state.py

```diff
@@ -505,7 +507,7 @@ def reconfigure(self, version: DeploymentVersion) -> bool:
                 deployment_config.user_config
             )
             self._ready_obj_ref = self._actor_handle.reconfigure.remote(
-                deployment_config
+                deployment_config, self._ready_obj_ref
```

Comment:
I think this is not necessary because the controller won't call `reconfigure` until after `initialize_and_get_metadata` has returned